crates/ruma-client-api/src/account/register.rs

```diff
@@ -96,7 +106,36 @@ pub mod v3 {
             /// ID of the registered device.
             ///
             /// Will be the same as the corresponding parameter in the request, if one was specified.
+            ///
+            /// Required if the request's `inhibit_login` was set to `false`.
             pub device_id: Option<OwnedDeviceId>,
+
+            /// A refresh token for the account.
+            ///
+            /// This token can be used to obtain a new access token when it expires by calling the
+            /// `/refresh` endpoint.
+            ///
+            /// Omitted if the request's `inhibit_login` was set to `true`.
+            #[cfg(feature = "unstable-msc2918")]
+            #[serde(skip_serializing_if = "Option::is_none")]
+            pub refresh_token: Option<String>,
+
+            /// The lifetime of the access token, in milliseconds.
+            ///
+            /// Once the access token has expired, a new access token can be obtained by using the
+            /// provided refresh token. If no refresh token is provided, the client will need to
+            /// re-login to obtain a new access token.
+            ///
+            /// If this is `None`, the client can assume that the access token will not expire.
+            ///
+            /// Omitted if the request's `inhibit_login` was set to `true`.
+            #[cfg(feature = "unstable-msc2918")]
+            #[serde(
+                with = "ruma_common::serde::duration::opt_ms",
+                default,
+                skip_serializing_if = "Option::is_none"
+            )]
+            pub expires_in_ms: Option<Duration>,
```

Comment:
nit: for language consistency, maybe this could be `expires_in`, with `expires_in_ms` being the serde-rename name?

Ditto below

Reply:
You actually made me realize I forgot to use the unstable names for the fields...

It's weird the MSC only mentions an unstable name for the `refresh_token` field, not the others.

Reply:
Synapse's implementation was in line with the MSC (before they stabilized the endpoint), it didn't use unstable prefixes for the response.